core-bundle/tests/Search/Backend/Provider/TableDataContainerProviderTest.php

```diff
@@ -0,0 +1,135 @@
+
+        parent::tearDown();
+    }
+
+    public function testSupports(): void
+    {
+        $provider = new TableDataContainerProvider(
+            $this->mockContaoFramework(),
+            $this->createMock(ResourceFinder::class),
+            $this->createMock(Connection::class),
+            $this->createMock(RecordLabeler::class),
+            $this->createMock(AccessDecisionManagerInterface::class),
+        );
+
+        $this->assertTrue($provider->supportsType(TableDataContainerProvider::TYPE_PREFIX.'foobar'));
+        $this->assertFalse($provider->supportsType('foobar'));
+    }
+
+    public function testUpdateIndex(): void
+    {
+        $connection = $this->createInMemorySQLiteConnection(
+            [
+                new Table('tl_content', [
+                    new Column('id', Type::getType(Types::INTEGER)),
+                    new Column('type', Type::getType(Types::STRING)),
+                    new Column('text', Type::getType(Types::STRING)),
+                ]),
+                new Table('tl_news', [
+                    new Column('id', Type::getType(Types::INTEGER)),
+                    new Column('headline', Type::getType(Types::STRING)),
+                    new Column('teaser', Type::getType(Types::STRING)),
+                ]),
+            ],
+            [
+                'tl_content' => [
+                    [
+                        'id' => 1,
+                        'type' => 'text',
+                        'text' => '<p>This is <em>some</em> content.',
+                    ],
+                ],
+                'tl_news' => [
+                    [
+                        'id' => 2,
+                        'headline' => 'This is my news 2!',
+                        'teaser' => 'A great teaser!',
+                    ],
+                    [
+                        'id' => 3,
+                        'headline' => 'This is my news 3!',
+                        'teaser' => 'Another great teaser!',
+                    ],
+                ],
+            ],
+        );
+
+        $framework = $this->mockContaoFramework();
+
+        $fixturesDir = $this->getFixturesDir();
+        $resourceFinder = new ResourceFinder(Path::join($fixturesDir, 'table-data-container-provider'));
+        $locator = new FileLocator(Path::join($fixturesDir, 'table-data-container-provider'));
+
+        $container = $this->getContainerWithContaoConfiguration($fixturesDir);
+        $container->set('contao.resource_finder', $resourceFinder);
+        $container->set('contao.resource_locator', $locator);
+
+        System::setContainer($container);
+
+        $provider = new TableDataContainerProvider(
+            $framework,
+            $resourceFinder,
+            $connection,
+            $this->createMock(RecordLabeler::class),
+            $this->createMock(AccessDecisionManagerInterface::class),
+        );
+
+        $documents = $provider->updateIndex(new UpdateAllProvidersConfig());
```

Comment:
```suggestion
        $documentsIterator = $provider->updateIndex(new UpdateAllProvidersConfig());
```

This will make it easier for the IDE to understand line 119.